### PR TITLE
Run the background worker during worker mode.

### PR DESCRIPTION
**Pull request by clokep**
This is a corresponding PR to matrix-org/synapse#8369 which runs an additional worker for background tasks. I suspect this won't really have impact on sytest since theoretically these shouldn't interfere with tests.

### CODE REVIEW

**Comment by clokep** [comment]
I think we would need to manually rebuild the docker image, CI doesn't seem to do that automatically?

**Comment by clokep** [comment]
This will conflict with #945, but hopefully that merges first.

File: lib/SyTest/Homeserver/Synapse.pm [hunk 1]
```diff
@@ -253,6 +253,7 @@ sub start
         send_federation       => ( not $self->{dendron} ),
         update_user_directory => ( not $self->{dendron} ),
         enable_media_repo     => ( not $self->{dendron} ),
+        run_background_tasks_on  => ( $self->{dendron} ? "background_worker1" : "master" ),
```
Comment (clokep) [inline lib/SyTest/Homeserver/Synapse.pm:256]: I'm not fully confident on what `self->{dendron}` is, but it seems to be only exist when workers are being used.

Comment (erikjohnston) [reply lib/SyTest/Homeserver/Synapse.pm:256]: Yeah, its basically the thing that enables `pydron` and stuff

**Comment by clokep** [comment]
@erikjohnston Merged in develop, I think this is ready again.

**Review by erikjohnston — APPROVED** [review]

**Pull request merged**